### PR TITLE
questions

### PR DESCRIPTION
**Pull request by katumwalauren**

### CODE REVIEW

File: src/exceptions/NotFoundException.ts [hunk 1]
```diff
@@ -2,5 +2,6 @@ export default class NotFoundException extends Error {
   constructor(message: string) {
```
Comment (katumwalauren) [inline src/exceptions/NotFoundException.ts:2]: Why is the message parameter typed as a string?

File: src/exceptions/NotFoundException.ts [hunk 2]
```diff
@@ -2,5 +2,6 @@ export default class NotFoundException extends Error {
...
     super(message);
     this.name = "NotFoundException";
```
Comment (katumwalauren) [inline src/exceptions/NotFoundException.ts:4]: How does extending the Error class benefit custom exception classes?
How would you improve  this NotFoundException class for a larger application
What does this.name = "NotFoundException"; do?

**Review by katumwalauren — COMMENTED** [review]
the questions are for files ParseException.ts and NotFoundException.ts

File: src/exceptions/NotFoundException.ts (hunk 1 above)
Comment (epicallan) [reply src/exceptions/NotFoundException.ts:2]: If you look at the documentation of the Error Class in NodeJS https://nodejs.org/api/errors.html#class-error you will see that it has a message field/property. The message property in an `Error` class is the text that is displayed when an error is thrown or logged. Therefore, when we extend `Error` and set the message to a new value, we are essentially providing new text for throwing or logging when using this new custom Error class.

File: src/exceptions/NotFoundException.ts (hunk 2 above)
Comment (epicallan) [reply src/exceptions/NotFoundException.ts:4]: > How does extending the Error class benefit custom exception classes?

We can only use instances or extensions of the Error class within try-catch blocks. Therefore, if we wish to work with custom exceptions that can be treated like native NodeJS Exceptions and Errors we need to extend the Error class. 

>What does this.name = "NotFoundException"; 

If you look at the documentation of the Error Class in NodeJS https://nodejs.org/api/errors.html#class-error, you will find that the Error class has a name property. The name property is typically part of the logged content when an Error is thrown. So, it's good practice to set it to a reasonable value.

**Comment by epicallan** [comment]
This PR should never have been merged

**Pull request merged**